pkg/processor/runtime/rpc/abstract.go

```diff
@@ -213,6 +214,40 @@ func (r *AbstractRuntime) SupportsRestart() bool {
 	return true
 }
 
+func (r *AbstractRuntime) Signal(signal syscall.Signal) error {
+
+	// signal and wait for process termination
+	if err := r.signal(signal); err != nil {
+		return errors.Wrap(err, "Failed to signal wrapper process")
+	}
+
+	// wait for process to finish or timeout
+	r.waitForProcessTermination(r.configuration.WorkerTerminationTimeout)
```

Comment:
you assume the signal terminates the process here, perhaps you want to move this functionality to the callee or provide "Terminate" function which signals with SIGTERM and waits, both can happen but surely "Signal" does not have to wait for worker termination in term of API